### PR TITLE
New Gulp Spice

### PR DESCRIPTION
**Pull request by jagtalon**
It uses http://npmsearch.com/ as the API provider which is what http://gulpjs.com/plugins/ uses.

screen shot 2016-01-15 at 6 01 09 am

---

https://duck.co/ia/view/gulp

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/spice/gulp/gulp.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                        return;
+                    }
+                    item.license = item.license || [];
+                    
+                    
+                    // Make sure that the query even exists in the name or description.
+                    if(!query_re.test(item.name[0]) && !query_re.test(item.description[0]) && 
+                       !query_re.test(item.keywords.toString())) {
+                        return;
+                    }
```
Comment (MrChrisW) [inline share/spice/gulp/gulp.js]: Maybe a relevancy block could be used (DDH.isRelevant) ? although this works :+1:

**Comment by MrChrisW** [comment]
Gulp! :+1:

File: share/spice/gulp/gulp.js (hunk 1 above)
Comment (jagtalon) [inline share/spice/gulp/gulp.js]: @MrChrisW relevancy block is better! I'll do that. :) I was being lazy haha

File: lib/DDG/Spice/Gulp.pm [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package DDG::Spice::Gulp;
+
+use DDG::Spice;
+
+spice is_cached => 1;
+spice wrap_jsonp_callback => 1;
+spice to => 'http://npmsearch.com/query?fields=name,keywords,rating,description,author,modified,homepage,version,license&q=$1%20AND%20keywords:gulpplugin&sort=rating:desc';
+
+triggers any => 'gulp plugin', 'gulp plugins', 'gulpjs', 'gulpjs plugin', 'gulpjs plugins', 'gulp.js', 'gulp.js plugin', 'gulp.js plugins';
```
Comment (moollaza) [inline lib/DDG/Spice/Gulp.pm]: I feel like we should generate this list?

```perl
my @triggers = map { "$_ plugin", "$_ plugins" } qw (gulp gulpjs gulp.js);
```

I think that works? I haven't tested it...

File: share/spice/gulp/gulp.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                        licenses: item.license.join(', ') || '',
+                        time: moment(item.modified[0], "YYYYMMDD").fromNow()
+                    };
+                },
+                templates: {
+                    group: 'text',
+                    detail: false,
+                    item_detail: false,
+                    variants: {
+                        tile: 'basic1',
```
Comment (moollaza) [inline share/spice/gulp/gulp.js]: You can remove this :+1:

File: share/spice/ruby_gems/footer.handlebars [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {{#if licenses}}
-<div>{{licenses}} Licence</div>
+<div>{{licenses}} License</div>
```
Comment (moollaza) [inline share/spice/ruby_gems/footer.handlebars]: :+1:

File: lib/DDG/Spice/Gulp.pm (hunk 2 above)
Comment (jagtalon) [inline lib/DDG/Spice/Gulp.pm]: @moollaza good idea! I'll try it

File: share/spice/gulp/gulp.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                        tileTitle: '1line',
+                        tileFooter: '2line',
+                        tileSnippet: 'large'
```
Comment (moollaza) [inline share/spice/gulp/gulp.js]: Seems like `small` would suffice here too.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=gulpjs sass)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/gulp_mobile.png)

**Comment by jagtalon** [comment]
@MrChrisW @moollaza ready for another review! I'm unsure about using the relevancy block for filtering search though--same issue as with Grunt: https://github.com/duckduckgo/zeroclickinfo-spice/pull/2414#issuecomment-172012637

I'm not sure if I'm missing something though

**Comment by jagtalon** [comment]
@moollaza I didn't use 'small' tile variant because it got too crowded. There needs to be some breathing room.

File: lib/DDG/Spice/Gulp.pm (hunk 2 above)
Comment (jagtalon) [inline lib/DDG/Spice/Gulp.pm]: @moollaza can't seem to get it work. I could do something like `map { ("$_ plugin", "$_ plugins") } qw (gulp gulpjs gulp.js);`, but I think you'd have to flatten the array after so might not be worth it.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=gulpjs sass)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/gulp_mobile.png)

**Comment by jagtalon** [comment]
Thanks, @daxtheduck ;)

File: share/spice/gulp/gulp.js (hunk 1 above)
Comment (MrChrisW) [inline share/spice/gulp/gulp.js]: @jagtalon No problem! To be honest I'm not the biggest fan of the current implementation of relevancy blocks in Spices!

Comment (jagtalon) [inline share/spice/gulp/gulp.js]: @MrChrisW agree--high friction at the moment and hard to debug. :)

**Comment by jagtalon** [comment]
@MrChrisW @moollaza moved a bunch of stuff to the relevancy block, but due to https://github.com/duckduckgo/zeroclickinfo-spice/pull/2414#issuecomment-172012637, I don't think I can move everything.

Ready for review!

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=gulpjs sass)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/gulp_mobile.png)

**Comment by moollaza** [comment]
@jagtalon Thanks! This LGTM :+1:

**Comment by jagtalon** [comment]
Yeaaaaaa

**Pull request merged**